DataBiosphere/Saturn-UI
# Reuse a Cromwell data disk if it exists, and fix spurious warnings about multiple disks [BW-897, BW-931].

When creating a Cromwell App, if a disk already exists from a previously deleted Cromwell app in that workspace, reuse it.

This also fixes a bug where if one tries to create a Galaxy instance in a workspace where an unattached disk originally created by Cromwell exists, then we attempt to reuse that Cromwell disk for Galaxy.

NOTE: to test this ticket you must test against dev.

To test BW-897:
1) Create a Workspace
2) Create a Crowell instance
3) Create a Galaxy instance.
4) Delete the Cromwell instance leaving the disk behind.
5) Delete the Galaxy instance, leaving the disk behind.
6) Create new Galaxy and Cromwell instances and verify they attach to the correct pre-existing disks (looking at information displayed in Cloud Environments/clusters page).

To test BW-931:
It is easy to verify that we don't show the warning when you have both a Jupyter disk and a Cromwell disk in the same project. To test that we DO show the warning if you > 1 disk for a given app type is somewhat harder. To test that, I removed the modification to CromwellModal so that we always create a new disk instance when creating a Cromwell instance. Then I could get 2 Cromwell disks in the same workspace, and I verified that we correctly show the warning. (Note: in this screenshot there is also a Galaxy disk left over from a deleted Galaxy instance… so 1 Jupyter, 1 Galaxy, and 2 Cromwell).

![image](https://user-images.githubusercontent.com/484484/143278012-c79a1e7f-b8ae-473c-80c4-dc95694eafd1.png)
